src/parser/shadowlands/modules/items/dungeons/SoullettingRuby.tsx

```diff
@@ -0,0 +1,297 @@
+   */
+  readonly critBudget: Readonly<{
+    base: number;
+    dynamic: number;
+  }> = {
+    base: -1,
+    dynamic: -1,
+  };
+  readonly item: Item;
+
+  healedAmount = 0;
+  /** Each entry is the crit value of a particular instance of the buff */
+  readonly casts: Array<{
+    timestamp: number;
+    targetName: ReactNode;
+    // Is it possible to get the enemys name here? To be able to display all casts
+    targetHpPercent: number;
+    buffValue?: {
+      base: number;
+      dynamic: number;
+      total: number;
+    };
+  }> = [];
+
+  constructor(
+    options: Options & {
+      abilities: Abilities;
+      buffs: Buffs;
+      castEfficiency: CastEfficiency;
+      enemies: Enemies;
+      statTracker: StatTracker;
+    },
+  ) {
+    super(options);
+
+    this.item = this.selectedCombatant.getItem(ITEMS.SOULLETTING_RUBY.id)!;
+    if (this.item == null) {
+      this.active = false;
+      return;
+    }
+
+    this.critBudget = calculateCritRating(this.item.itemLevel);
+
+    // This value needs to update with every use of the trinket, we
+    // just set it to the lowest possible value to start with.
+    options.statTracker.add(BUFF.id, {
+      crit: this.critBudget.base,
+    });
+
+    // Add buff to show up in timeline
+    options.buffs.add({
+      spellId: BUFF.id,
+      timelineHighlight: true,
+      triggeredBySpellId: CAST.id,
+    });
+
+    // Add cast as an ability to show effective usage and cooldown in timeline
+    options.abilities.add({
+      spell: CAST.id,
+      category: Abilities.SPELL_CATEGORIES.ITEMS,
+      cooldown: COOLDOWN_SECONDS,
+      gcd: null,
+      castEfficiency: {
+        suggestion: true,
+        recommendedEfficiency: 0.8,
+      },
+    });
+
+    this.addEventListener(Events.cast.spell(CAST).by(SELECTED_PLAYER), this.onCast);
+    this.addEventListener(Events.heal.spell(HEAL).to(SELECTED_PLAYER), this.onHeal);
+  }
+
+  /** When cast, figure out the multiplier of the buff we will gain later */
+  onCast(event: CastEvent) {
+    if (!HasTarget(event)) {
+      console.error('SoullettingRuby: cast has no target');
+      return;
+    }
+
+    const { hitPointsPercent, enemy } = this.enemiesHealth.getHealthEnemy(event) || {};
+    const targetName = enemy?.name ?? <span className="poor">Unknown</span>;
+
+    if (hitPointsPercent == null) {
+      this.casts.push({
+        timestamp: event.timestamp,
+        targetName,
+        targetHpPercent: -1,
+        // Since we don't know the buff value, we dont store it
+      });
+
+      // Update statTracker with lowest possible value
+      this.updateBuffValue(this.calculateBuffValue(1));
+    } else {
+      const buffValue = this.calculateBuffValue(hitPointsPercent);
+
+      this.casts.push({
+        timestamp: event.timestamp,
+        targetName,
+        targetHpPercent: hitPointsPercent,
+        buffValue,
+      });
+
+      this.updateBuffValue(buffValue);
```

Comment:
don't act on this comment, but if they add more of these variable-value trinkets with no stacks or other tracking info, we should probably change how we handle this.

the stat tracker can have a function per stat, but the function currently doesn't include the parameters that you'd need to distinguish different uses of the trinket (so we'd need core changes to support that). having the function receive e.g. the `BuffEvent` that then can be correlated with a value via closing over the analyzer state is cleaner.

Reply:
Not sure of how to connect the buffevent with the appropiate castevent in a hypotethical case similiar to this. But being able to have more information in the callback would definitely be a good thing. 

They should just god damn log the strength of the buff tho.......